master/buildbot/steps/artifact.py

```diff
@@ -235,6 +235,34 @@ def start(self):
         self.setCommand(command)
         ShellCommand.start(self)
 
+
+def checkWindowsSlaveEnvironment(step, key):
+    return key in step.build.slavebuilder.slave.slave_environ.keys() \
+           and step.build.slavebuilder.slave.slave_environ[key] == 'Windows_NT'
+
+
+def _isWindowsSlave(step):
+        slave_os = step.build.slavebuilder.slave.os and step.build.slavebuilder.slave.os == 'Windows'
+        slave_env = checkWindowsSlaveEnvironment(step, 'os') or checkWindowsSlaveEnvironment(step, 'OS')
+        return slave_os or slave_env
+
+
+def retryCommandLinuxOS(command):
+    return 'n=0; false; until [[ $? -eq 0 || $n -ge 5 ]]; do n=$[$n+1]; sleep 5; ' + command + "; done"
+
+
+def retryCommandWindowsOS(command):
+    return 'for /L %%i in (1,1,5) do (sleep 5 & ' + command + ' && exit 0)'
```

Comment:
Why not do this with powershell and make peoples lifes easier? I doubt there are many windows XP agents out there.